src/Runner.Worker/JobRunner.cs

```diff
@@ -130,6 +130,10 @@ public async Task<TaskResult> RunAsync(AgentJobRequestMessage message, Cancellat
                 Directory.CreateDirectory(toolsDirectory);
                 jobContext.SetRunnerContext("tool_cache", toolsDirectory);
 
+                // Extract from extra claims
+                string runnerEnvironment = "github-hosted"
+                jobContext.SetRunnerContext("environment", runnerEnvironment);
```

Comment:
The runner env was added to the OIDC claims: https://github.com/github/actions-dotnet/pull/14382 but I don't know how to actually get at this value here.